src/ImageSharp/Processing/Processors/Convolution/Convolution2PassProcessor{TPixel}.cs

```diff
@@ -64,44 +59,364 @@ protected override void OnFrameApply(ImageFrame<TPixel> source)
             // for source and target bulk pixel conversion.
             var operationBounds = new Rectangle(interest.X, interest.Y, interest.Width * 2, interest.Height);
 
-            using (var mapX = new KernelSamplingMap(this.Configuration.MemoryAllocator))
+            // We can create a single sampling map with the size as if we were using the non separated 2D kernel
+            // the two 1D kernels represent, and reuse it across both convolution steps, like in the bokeh blur.
+            using var mapXY = new KernelSamplingMap(this.Configuration.MemoryAllocator);
+
+            mapXY.BuildSamplingOffsetMap(this.Kernel.Length, this.Kernel.Length, interest);
```

Comment:
I was initially concerned about the increased size of the map but doing the math I think it's acceptable.

Reply:
From what I see in the actual map code, the allocations are the same as before, just done at the same time, no?
As in, looking at the allocation bit in the `KernelSamplingMap` class:

https://github.com/SixLabors/ImageSharp/blob/cd1a25efeeb4b243cbfcfff8950fc215523077b7/src/ImageSharp/Processing/Processors/Convolution/KernelSamplingMap.cs#L44-L45

I did this change mostly just to avoid the extra computations for the horizontal/vertical maps that would not be used at all when doing either the horizontal or vertical passes for the specialized 1D convolution processors 😄

Reply:
Haha... yeah you're right! Awesome.